### ISSUE TITLE
Error:Could not find best line for multiline plate

### ISSUE THREAD

**Issue by astrung** [openalpr/openalpr#327]
By some images,it only detects for one-line on plates.If i config for multi-line,by turn on debug,i get this error:Could not find best line for multiline plate.By tracing, i see in `linefinder.cpp` , `std::vector<cv::Point> findNextBestLine(cv::Size imageSize, std::vector<cv::Point> bestLine);` function can not get an other line in plates.
Continuing to dig more , in `HistogramHorizontal.cpp` , `std::vector<std::pair<int, int> > get1DHits(int yOffset)` function return only 3 elements with error images >> I think it is a point to fix it.
Can anyone suggest to fix it??
This is error images:
![4](https://cloud.githubusercontent.com/assets/9162294/14705048/eeb5c538-07e1-11e6-90be-5b4d2e858bc0.jpeg)

**Comment by matthill** [comment]
Can you upload the detector and country config you're using?

**Comment by astrung** [comment]
@matthill :
https://drive.google.com/file/d/0Bwo9D-C40V-1MVltRzhfUVYtWnM/view?usp=sharing

https://drive.google.com/file/d/0Bwo9D-C40V-1REFtaEFfeHI1Wmc/view?usp=sharing

**Comment by astrung** [comment]
@matthill Do you have any suggestions ??

**Comment by astrung** [comment]
i have found that with different options in config.conf , we will get variant histogram images.
With above image,by this config:
`template_max_width_px = 120`
`template_max_height_px = 60`
i get this histogram image.It have three white spaces,and `std::vector<std::pair<int, int> > get1DHits(int yOffset)` return only 3 points.
![fail](https://cloud.githubusercontent.com/assets/9162294/14841236/16d83d24-0c66-11e6-96b8-9b0e6c91a352.jpg)
And by this config:
`template_max_width_px = 190`
`template_max_height_px = 150`
It return a different histogram image . It have four separate whites spaces , and `std::vector<std::pair<int, int> > get1DHits(int yOffset)` return  4 points ( correct !!! )
![success](https://cloud.githubusercontent.com/assets/9162294/14841291/a30151d2-0c66-11e6-8a92-840d9c397654.jpg)
So,how to config for there options ?? ( `template_max_width_px` and `template_max_height_px`  )

**Comment by matthill** [comment]
I changed the character height steps to these values:
; 35-50; 45-60, 55-70, 65-80, 75-90
char_analysis_min_pct = 0.25
char_analysis_height_range = 0.10
char_analysis_height_step_size = 0.7
char_analysis_height_num_steps = 5


And it seems to work better.  The characters are all segmented correctly.  It mistakes the "2"s for "Z"s but that can be fixed with OCR training:

plate0: 10 results
    - 61T3-ZZZ2	 confidence: 88.0903
    - 6T3I-ZZZ2	 confidence: 87.2305
    - 6T3-ZZZ2	 confidence: 86.9019
    - 6T31-ZZZ2	 confidence: 86.7999
    - 61T3-Z2Z2	 confidence: 86.0639
    - 6T3I-Z2Z2	 confidence: 85.2042
    - 6T3-Z2Z2	 confidence: 84.8756
    - 6T3D-ZZZ2	 confidence: 84.8713
    - 6T31-Z2Z2	 confidence: 84.7736
    - 6TT3-ZZZ2	 confidence: 83.8774

**Comment by astrung** [comment]
Can you explain your changed config ?? It isn't explained in document.

**Comment by matthill** [comment]
Yes, you're correct.  This step is what looks for character "blobs" to find a line of text.  The "min_pct" is the vertical height percent to start searching (in this case 25% of the plate height).  It looks in a range between 25% and 35% (.25 + .10) of the plate height.  Then it increases 5 times (num_steps) by 7% (step_size).  So, the next step would be 32% - 42%.  Whichever band has the best character match is matched as a line.

Since you copied from the single-line EU config, it wouldn't work, since those values were looking for a single line on a plate with very small height (the characters took up most of the height).  In your two-line plate, the characters couldn't possibly take up more than 50% of the plate height (since there's 2 lines)

**Comment by ozkankaya** [comment]
astrung and matthill,
It is a good job. In Turkey, has both 1 and 2 lines plates. Can you send or publish your detect and county.conf files as example? I tried to create these files a lot of times. But it wasn't worked. I want to grow this project for Turkey. I am waiting your helps.
